backpack-compose/src/main/kotlin/net/skyscanner/backpack/compose/chip/BpkChip.kt

```diff
@@ -0,0 +1,162 @@
+/*
+ * Backpack for Android - Skyscanner's Design System
+ *
+ * Copyright 2018 Skyscanner Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.skyscanner.backpack.compose.chip
+
+import androidx.compose.animation.animateColorAsState
+import androidx.compose.foundation.background
+import androidx.compose.foundation.layout.Arrangement
+import androidx.compose.foundation.layout.Row
+import androidx.compose.foundation.layout.height
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.selection.selectable
+import androidx.compose.foundation.shape.RoundedCornerShape
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.getValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.clip
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.graphics.takeOrElse
+import androidx.compose.ui.text.style.TextOverflow
+import net.skyscanner.backpack.compose.icon.BpkIcon
+import net.skyscanner.backpack.compose.icon.BpkIconSize
+import net.skyscanner.backpack.compose.text.BpkText
+import net.skyscanner.backpack.compose.theme.BpkTheme
+import net.skyscanner.backpack.compose.tokens.BpkColor
+import net.skyscanner.backpack.compose.tokens.BpkSpacing
+import net.skyscanner.backpack.compose.tokens.CloseCircle
+import net.skyscanner.backpack.compose.tokens.Tick
+import net.skyscanner.backpack.compose.utils.dynamicColorOf
+
+enum class BpkChipState {
```

Comment:
To match other platforms & the existing chip (as well as other components, like buttons, toggleable components etc) i'd separate this to two boolean flags of selected & enabled. while currently the style for a disabled chip is separate this might not be the case going forward

Reply:
The UI representation for selected & unselected, but disabled is currently the same. On the other hand in talk back the announcement separates selected & disabled (and announces both). All platforms currently provide boolean flags for the two, so if we do add a UI distinction for selected & unselected + disabled we will be able to add that without API changes on all platforms, but compose.